packages/rum/src/boot/recorderApi.ts

```diff
@@ -236,4 +88,81 @@ export function makeRecorderApi(
     getReplayStats: (viewId) =>
       getDeflateWorkerStatus() === DeflateWorkerStatus.Initialized ? getReplayStatsImpl(viewId) : undefined,
   }
+
+  function onRumStart(
+    lifeCycle: LifeCycle,
+    configuration: RumConfiguration,
+    sessionManager: RumSessionManager,
+    viewHistory: ViewHistory,
+    worker: DeflateWorker | undefined
+  ) {
+    if (configuration.startSessionReplayRecordingManually) {
+      state.setStatus(RecorderStatus.Stopped)
+    }
+    lifeCycle.subscribe(LifeCycleEventType.SESSION_EXPIRED, () => {
+      if (state.status === RecorderStatus.Starting || state.status === RecorderStatus.Started) {
+        strategies.stopStrategy()
+        state.setStatus(RecorderStatus.IntentToStart)
+      }
+    })
+
+    // Stop the recorder on page unload to avoid sending records after the page is ended.
+    lifeCycle.subscribe(LifeCycleEventType.PAGE_EXITED, (pageExitEvent) => {
+      if (pageExitEvent.reason === PageExitReason.UNLOADING) {
+        strategies.stopStrategy()
+      }
+    })
+
+    lifeCycle.subscribe(LifeCycleEventType.SESSION_RENEWED, () => {
+      if (state.status === RecorderStatus.IntentToStart) {
+        strategies.startStrategy()
+      }
+    })
+
+    let cachedDeflateEncoder: DeflateEncoder | undefined
+
+    function getOrCreateDeflateEncoder() {
+      if (!cachedDeflateEncoder) {
+        worker ??= startDeflateWorker(
+          configuration,
+          'Datadog Session Replay',
+          () => {
+            strategies.stopStrategy()
+          },
+          createDeflateWorkerImpl
+        )
+
+        if (worker) {
+          cachedDeflateEncoder = createDeflateEncoder(configuration, worker, DeflateEncoderStreamId.REPLAY)
+        }
+      }
+      return cachedDeflateEncoder
+    }
+
+    strategies = createPostStartStrategy(
+      configuration,
+      lifeCycle,
+      sessionManager,
+      viewHistory,
+      startRecordingImpl,
+      getOrCreateDeflateEncoder,
+      state
+    )
+
+    if (state.status === RecorderStatus.IntentToStart) {
+      strategies.startStrategy()
+    }
+  }
+}
+
+export function createPreStartStrategy(state: RecorderState): Strategy {
```

Comment:
Trivial nit: from a code organization perspective, I feel like if I see that `createPostStartStrategy()` lives in `postStartStrategy.ts`, I might expect this function to live in `preStartStrategy.ts`. Moving it there might imply putting `RecorderState`, `RecorderStatus`, and `Strategy` in a separate file as well, but again doing that might make it a bit easier to find them. Definitely not a strongly held view if you prefer this way of organizing things, though!

Reply:
Done!